File: client/src/app/_models/produto/livro.ts
import { Autor } from "../autor";
import { Editora } from "../editora";
import { Fornecedor } from "../fornecedor";
import { Categoria } from "./categoria";
import { Usuario } from "./usuario";

export interface Livro {
    id: string;
    titulo: string;
    imagem: string;
    livroImagem: string;
    descricao: string;
    categoriaId: string;
    autorId: string;
    editoraId: string;
    fornecedorId: string;
    estado: boolean;
    dataCadastro: Date;
    dataAtualizacao: Date;

    categoria: Categoria;
    autor: Autor;
    fornecedor: Fornecedor;
    editor: Editora;
    usuario: Usuario;
}

File: src/VirtualLibrary.Domain/Models/Livro.cs
using System;
using System.Collections.Generic;

namespace VirtualLibrary.Domain.Models
{
    public class Livro : Entity
    {
        public Guid? UsuarioId { get; set; }
        public Guid CategoriaId { get; set; }
        public Guid AutorId { get; set; }
        public Guid EditoraId { get; set; }
        public Guid FornecedorId { get; set; }

        public string Titulo { get; set; }
        public string Imagem { get; set; }
        public string Descricao { get; set; }
        public bool Estado { get; set; }
        public DateTime DataCadastro { get; set; }
        public DateTime DataAtualizacao { get; set; }

        /*EF Core - Relations*/
        public Usuario Usuario { get; set; }
        public Categoria Categoria { get; set; }
        public Autor Autor { get; set; }
        public Editora Editora { get; set; }
        public Fornecedor Fornecedor { get; set; }

        public IEnumerable<Aluguer> Alugueres { get; set; }
    }
}